File: 0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int area=0,max=0;
        while(l<r)
        {
            if(height[l]<height[r])
            {
                area=height[l]*(r-l);
                l++; 
            }
            else
            {
                area=height[r]*(r-l);
                r--;
            }
            if(area>max)
            {
                max=area;
            }
        }
        return max;
    }
}

File: 0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        max_a=0
        while l<r:
            cur_h=min(height[l],height[r])
            d=cur_h*(r-l)
            max_a=max(d,max_a)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return max_a
